node/silkworm/stagedsync/stage_log_index.cpp

```diff
@@ -215,4 +215,59 @@ StageResult unwind_log_index(TransactionManager& txn, const std::filesystem::pat
     return StageResult::kSuccess;
 }
 
+void prune_log_index(TransactionManager& txn, etl::Collector& collector, uint64_t prune_from,
+                                 bool topics) {
+    auto last_processed_block{db::stages::get_stage_progress(*txn, db::stages::kLogIndexKey)};
+
+    auto index_table{topics ? db::open_cursor(*txn, db::table::kLogTopicIndex)
+                    : db::open_cursor(*txn, db::table::kLogAddressIndex)};
+
+    if (index_table.to_first(/* throw_notfound = */ false)) {
+        auto data{index_table.current()};
+        while (data) {
+            // Get bitmap data of current element
+            auto key{db::from_slice(data.key)};
+            auto bitmap_data{db::from_slice(data.value)};
+            auto bm{roaring::Roaring::readSafe(byte_ptr_cast(bitmap_data.data()), bitmap_data.size())};
+            // Check wheter we should skip the current bitmap
+            if (bm.minimum() >= prune_from) {
+                data = index_table.to_next(/*throw_notfound*/ false);
+                continue;
+            }
+            // check if prune can be applied
+            if (bm.maximum() >= prune_from) {
+                // Erase elements that are below prune_from
+                bm &= roaring::Roaring(roaring::api::roaring_bitmap_from_range(prune_from, last_processed_block + 1, 1));
+                Bytes new_bitmap(bm.getSizeInBytes(), '\0');
+                bm.write(byte_ptr_cast(&new_bitmap[0]));
+                // generates new key
+                Bytes new_key(key.size(), '\0');
+                std::memcpy(&new_key[0], key.data(), key.size());
+                endian::store_big_u32(&new_key[new_key.size() - 4], UINT32_MAX);
```

Comment:
What's the logic with `UINT32_MAX` here?

Reply:
"mis-pasting" on my part and i did not include a case for it

Reply:
now i fixed it

Reply:
Well, `UINT32_MAX` is still here. If it's correct, please add more comments - otherwise the logic is not immediately clear.